packages/vite/src/shared/moduleRunnerTransport.ts

```diff
@@ -33,7 +33,15 @@ type InvokeableModuleRunnerTransport = Omit<ModuleRunnerTransport, 'invoke'> & {
 }
 
 function reviveInvokeError(e: any) {
-  return Object.assign(new Error(e.message || 'Unknown invoke error'), e)
+  const innerError = new Error('received error stacktrace')
+  innerError.stack = e.stack
+
+  // set properties to wrapped error, but use the current stacktrace
+  const wrappedError = new Error(e.message || 'Unknown invoke error', {
+    cause: innerError,
+  })
+  Object.assign(wrappedError, { ...e, stack: wrappedError.stack })
```

Comment:
Overall fixes make sense, but I'm not sure which stack should be visible to users. Technically the call flow is in a following order:

```
environment import -> runner invoke -> environment fetchModule -> plugin error
```

so having plugin errors location at the top of the stack seems natural. I think this was also the case on Vite 5 ssr and user plugin's stack can be important for debugging.

As an alternative, what about injecting runner side stack as a separate property like `e.runnerStack = wrappedError.stack`? That should allow `console.log(e)` to print out the runner stack.

Reply:
I think the order itself is natural as this is how the cause property works. But you have a point. Most users would want to know where the plugin error happened. I'll put it in a different property.

Reply:
Updated in edb34684dd3f6aefa215d663df11fb7b86c024dc
I passed the whole error to the property, otherwise the console.log will output with quotes and escapes.
```
Error: foo
    at file:///D:/documents/GitHub/vite/foo.mjs:2:22
    at ModuleJob.run (node:internal/modules/esm/module_job:268:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:543:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5) {
  runnerStack: 'Error: RunnerError\n' +
    '    at file:///D:/documents/GitHub/vite/foo.mjs:3:47\n' +
    '    at ModuleJob.run (node:internal/modules/esm/module_job:268:25)\n' +
    '    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:543:26)\n' +
    '    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)'
}
```
```
Error: foo
    at file:///D:/documents/GitHub/vite/foo.mjs:2:22
    at ModuleJob.run (node:internal/modules/esm/module_job:268:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:543:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5) {
  runnerStack: Error: RunnerError
      at file:///D:/documents/GitHub/vite/foo.mjs:3:47
      at ModuleJob.run (node:internal/modules/esm/module_job:268:25)
      at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:543:26)
      at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)
}
```